### ISSUE TITLE
Elm-make fails seemingly at random

### ISSUE THREAD

**Issue by szabba** [elm-lang/elm-compiler#1235]
I get the following error a couple times a day

```bash
$ elm-make SomeModule.elm
[                                                  ] - 0 / 1elm-make: the annotation added in 'depattern' should not be observed!
elm-make: thread blocked indefinitely in an MVar operation
```

Each time it appears, it repeats itself a couple times, but usually disappears _without_ changes in the code being compiled.

```bash
$ elm-make SomeModule.elm
Success! Compiled 0 modules.                                        
Successfully generated index.html
```

I'm using Elm 0.16.0 installed through the npm.
```bash
$ npm list -g elm
/usr/lib
└── elm@0.16.0
```

I'd like to be able to provide more details but I don't even know what to look into.

**Comment by szabba** [comment]
Note: this seems to have something to do with concurrency. The `linter-elm-make` extension in Atom stopped working for me at some point, but I didn't bother turning it off before. I'll try to report if this still happens with it disable.

**Comment by szabba** [comment]
This still happens on my machine with the Atom extension uninstalled. I know this report isn't very helpful. Is there anything I could do to improve it? I don't really know what other info I could provide nor where to get it from.

**Comment by mgold** [comment]
Sorry you're having trouble. I'm as stumped as you are. I don't think this will work, but you can try deleting `elm-stuff` which will cause the dependencies to be downloaded and compiled again. Just in case there's a bad intermediate file in the cache.

**Comment by evancz** [comment]
`elm-make` compiles files concurrently as much as possible, so if you are on a platform where other people are messing with files, it may interfere. So If you can say your OS, editor, environment, etc. we can probably help more. That said, ultimately, I think the creators of the tools that are messing with files should be the ones dealing with this issue. So I am going to close this particular issue, though it totally makes sense to continue discussion here. Once things get pinned down, it makes sense to (1) open an issue with the relevant tooling person or (2) open an issue here if there are concrete proposals on how `elm-make` should change to accommodate tools better.
